### PR TITLE
Add `advertisingSid` property to `ScanResult`

### LINKED ISSUE #854: Add `advertisingSid` property to `ScanResult` and `RxBleInternalScanResult`
**Is your feature request related to a problem? Please describe.**
`com.polidea.rxandroidble2.scan.ScanResult` does not contain an information about advertising sid which is needed for supporting Bluetooth Auracast feature (https://www.bluetooth.com/wp-content/uploads/Files/Specification/HTML/24670-BASS-html5/out/en/index-en.html#UUID-2a17a3a2-36f9-8cb4-0877-3b56055a7955). It means that there is no option to retrieve this information from the `rxandroidble` library.

**Describe the solution you'd like**
`com.polidea.rxandroidble2.scan.ScanResult` and `com.polidea.rxandroidble2.internal.scan.RxBleInternalScanResult` classes can be extended with `advertisingSid` property. This value can be retrieved using `android.bluetooth.le.ScanResult.getAdvertisingSid` method (https://developer.android.com/reference/kotlin/android/bluetooth/le/ScanResult#getadvertisingsid) and propagated in `com.polidea.rxandroidble2.internal.scan.InternalScanResultCreator`.

I can volunteer and submit a PR soon.

### PR DESCRIPTION
**Pull request by KrzysztofMamak**
Add `advertisingSid` property to `ScanResult`.

Fixes #854

### CODE REVIEW

**Comment by Albert221** [comment]
friendly ping @dariuszseweryn :)

File: rxandroidble/src/main/java/com/polidea/rxandroidble2/internal/scan/InternalScanResultCreator.java [hunk 1]
```diff
@@ -36,21 +37,23 @@ public InternalScanResultCreator(ScanRecordParser scanRecordParser, IsConnectabl
...
+        final Integer advertisingSid = Build.VERSION.SDK_INT >= Build.VERSION_CODES.O ? result.getAdvertisingSid() : null;
         return new RxBleInternalScanResult(result.getDevice(), result.getRssi(), result.getTimestampNanos(), scanRecord,
-                ScanCallbackType.CALLBACK_TYPE_BATCH, isConnectableChecker.check(result));
+                ScanCallbackType.CALLBACK_TYPE_BATCH, isConnectableChecker.check(result), advertisingSid);
     }
 
     @RequiresApi(21 /* Build.VERSION_CODES.LOLLIPOP */)
     public RxBleInternalScanResult create(int callbackType, ScanResult result) {
         final ScanRecordImplNativeWrapper scanRecord = new ScanRecordImplNativeWrapper(result.getScanRecord(), scanRecordParser);
+        final Integer advertisingSid = Build.VERSION.SDK_INT >= Build.VERSION_CODES.O ? result.getAdvertisingSid() : null;
```
Comment (dariuszseweryn) [inline rxandroidble/src/main/java/com/polidea/rxandroidble2/internal/scan/InternalScanResultCreator.java:54]: Could we have this logic injected in constructor?
Device SDK level is known at construction time and we don't need to calculate it for every result

File: rxandroidble/src/main/java/com/polidea/rxandroidble2/scan/ScanResult.java [hunk 2]
```diff
@@ -13,15 +13,17 @@ public class ScanResult {
     private final ScanCallbackType callbackType;
     private final ScanRecord scanRecord;
     private final IsConnectable isConnectable;
+    private final Integer advertisingSid;
 
     public ScanResult(RxBleDevice bleDevice, int rssi, long timestampNanos, ScanCallbackType callbackType,
-                      ScanRecord scanRecord, IsConnectable isConnectable) {
+                      ScanRecord scanRecord, IsConnectable isConnectable, Integer advertisingSid) {
```
Comment (dariuszseweryn) [inline rxandroidble/src/main/java/com/polidea/rxandroidble2/scan/ScanResult.java:19]: could we copy the constructor (telescope) and keep the original to not break the API? we could add `@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)`

**Review by dariuszseweryn — APPROVED** [review]
Awesome, thank you!

**Comment by Albert221** [comment]
@dariuszseweryn hey! Do you plan on releasing a version with that addition soon? :)

**Comment by dariuszseweryn** [comment]
As soon as possible. I seem to have a problem with recent configuration changes on Sonatype which prevent me from uploading an artifact. I'll need to reach their support probably

**Comment by dariuszseweryn** [comment]
1.19.0 should be available shortly

**Pull request merged**